Cython/Compiler/Nodes.py

```diff
@@ -4852,6 +4853,8 @@ def analyse_declarations(self, env):
         if self.doc_node:
             self.doc_node.analyse_target_declaration(cenv)
         self.body.analyse_declarations(cenv)
+        if hasattr(self.class_result, "analyse_class_annotations"):
```

Comment:
I prefer making methods generally available over testing for them with `hasattr()`. Is there a case that you ran into where the method was not there? The question is, at which level in the node class hierarchy we would need to introduce the method.

Reply:
Well, this method is only available for a `Py3ClassNode`, so I'm hedging my bets against this logic running against a `PyClassNode`, which doesn't have this logic... 

I've been thinking that we have a bit of a chicken-and-egg problem here, where we can't analyze the annotations without analyzing the class body, but this needs to be a part of the class body at the end of the analysis. Perhaps it would be better to have this logic live in `StatListNode.analyse_declarations()` and be activated if passed a `ClassScope` instance?

Reply:
Thinking more on this on - if we move this logic to `StatListNode`, it will pave the way for re-using this logic for module-scoped annotations with very minimal effort, since they both follow similar chains. Theoretically all we'd need to do then is get rid of the module scope bail out in `NameNode.declare_from_annotation()` and update the branch in `StatListNode.analyse_declarations()` to run on a module scope as well.

Reply:
… except that statement nodes can be wrapped in all sorts of other nodes, not just lined up in a `StatListNode`. No, I think in the end, we'll have to let the nodes with annotations inject themselves into the dict along the way of their execution.

We'll also have to make this work at some point:
```
>>> class C:
...   del __annotations__
...   a: int = 5
... 
>>> C
<class '__main__.C'>
>>> C.a
5
>>> C.__annotations__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: type object 'C' has no attribute '__annotations__'
```

Reply:
I've added a new issue to address this behavior and others, such as the custom namespace object: #3839

Reply:
As a safety measure, I added a no-op `analyse_annotations()` to `ClassNode` so that this check is no longer necessary.